src/node_messaging.cc

```diff
@@ -711,6 +711,13 @@ void MessagePort::PostMessage(const FunctionCallbackInfo<Value>& args) {
     return THROW_ERR_MISSING_ARGS(env, "Not enough arguments to "
                                        "MessagePort.postMessage");
   }
+  if (!args[1]->IsNullOrUndefined() && !args[1]->IsObject()) {
+    // Browsers also do not throw on `null` or objects, although it really
+    // should be an array or undefined, thus the mismatch between the checks
+    // above and the actual error message.
```

Comment:
The reason for this is due to the fact that browsers have an overload that accepts an option bag in addition to just a transfer list. See https://html.spec.whatwg.org/multipage/web-messaging.html#messageport. How browsers resolve the overloading is roughly something like this

1. If second arg is undefined or null then ignore.
2. If second arg is not an object (i.e., also not a function), then throw.
3. Try to get the Symbol.iterator property of the second arg.
   1. If that throws, then rethrow the exception.
   2. If that returns anything other than undefined or function, also throw an exception.
   3. If it returns a function, then treat this as an iterable and use the equivalent of Array.from to convert it to a list.
   4. Otherwise (it returns undefined), then treat the second arg as an option bag. Try to get the transferList property and then convert it to a list through something like Array.from if present.

Reply:
@TimothyGu I’m adding a TODO comment in this PR and merge this with the current state, if that’s okay with you

Reply:
Yeah sure